design/dsl/request_test.go

```diff
@@ -0,0 +1,131 @@
+package dsl_test
+
+import (
+	"testing"
+
+	"reflect"
+
+	"goa.design/goa.v2/design"
+	. "goa.design/goa.v2/design/dsl"
+	"goa.design/goa.v2/eval"
+)
+
+func TestRequest(t *testing.T) {
+	var RequestType = Type("Request", func() {
+		Description("Optional description")
+		Attribute("required", design.String)
+		Attribute("name", design.String)
+		Required("required")
+	})
+	cases := map[string]struct {
+		DSL         func()
+		Assert      func(t *testing.T, req design.DataType)
+		ExpectError bool
+	}{
+		"DefinedUserType": {
+			DSL: func() {
+				Request(RequestType, func() {
```

Comment:
@raphael Right, I tried using the code below, so not sure if it is a bug, but when I tried to do that I get nil.